### PR TITLE
Moths can now consume clothes again

### LINKED ISSUE #55347: Moths no longer get nutrition from clothing
Reporting client version: 513.1536

## Round ID:
[151522](https://scrubby.melonmesa.com/round/151522)

## Testmerges:
- It happened after newfood was introduced https://github.com/tgstation/tgstation/pull/55160

## Reproduction:
I joined as a moth and got hungry. Decided to snack on some shoes as is the norm and found that it neither reduced my hunger nor seemed to cause me to get full. I finished an entire pair of shoes without becoming full. I also ate a watermelon which filled me up (and made me sick) so it only seems to be related to moths eating clothes.

Also, usually when eating clothing moths get messages saying you are eating the clothes, however the only messages that came up were for the clothes slowly becoming shredded.

Normal food makes moths vomit so they kinda need to eat shoes again.

Reproduce by becoming moth and eating clothing when hungry.

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Fixes #55347.

Because the food eating is invoked async-ly, the `qdel` was being run while you were still taking a bite. This means it was no longer edible, thus quitting the check after the timer was finished.

This moves moths eating clothes behavior to instead lazily create the food item once chomped into, which is then preserved and reused until the clothing itself goes away.

## Why It's Good For The Game
meh

## Changelog
:cl:
fix: Clothes now consume themselves properly.
/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -76,21 +81,50 @@
...
 //This code is cursed, moths are cursed, and someday I will destroy it. but today is not that day.
 /obj/item/food/clothing
 	name = "temporary moth clothing snack item"
 	desc = "If you're reading this it means I messed up. This is related to moths eating clothes and I didn't know a better way to do it than making a new food object. <--- stinky idiot wrote this"
-	food_reagents = list(/datum/reagent/consumable/nutriment = 1)
+	bite_consumption = 1
+	// sigh, ok, so it's not ACTUALLY infinite nutrition. this is so you can eat clothes more than...once.
+	// bite_consumption limits how much you actually get, and the take_damage in after eat makes sure you can't abuse this.
+	// ...maybe this was a mistake after all.
+	food_reagents = list(/datum/reagent/consumable/nutriment = INFINITY)
```
Comment (Qustinnus) [inline code/modules/clothing/clothing.dm:93]: Crappy hack, I don't see why you need this if you use the callback.

**Comment by Qustinnus** [comment]
oranges hires too many feature coders

**Comment by brotherbeyondd** [comment]
thank fuck

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm:93]: Do you mean giving the nutrition on after_eat and having the clothing food itself have no nutrition? If so, I'm not sure how that's much better, and if not, I'd like to know what you mean by this.

Comment (Jared-Fogle) [reply code/modules/clothing/clothing.dm:93]: Floyd said he didn't care about this anymore on Discord.

**Review by Timberpoes — APPROVED** [review]
tl;dr maintainer merge machine go brrrrrrrr
